pkg/xray/resource_xray_watch_test.go

```diff
@@ -0,0 +1,565 @@
+package xray
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+var tempStructWatch = map[string]string{
+	"resource_name":     "",
+	"watch_name":        "xray-watch",
+	"description":       "This is a new watch created by TF Provider",
+	"active":            "true",
+	"watch_type":        "all-repos",
+	"filter_type_0":     "regex",
+	"filter_value_0":    ".*",
+	"filter_type_1":     "package-type",
+	"filter_value_1":    "Docker",
+	"policy_name_0":     "xray-policy-0",
+	"policy_name_1":     "xray-policy-1",
+	"watch_recipient_0": "test@email.com",
+	"watch_recipient_1": "test@email.com",
+}
+
+func TestAccWatch_allReposSinglePolicy(t *testing.T) {
+	_, fqrn, resourceName := mkNames("watch-", "xray_watch")
+	tempStruct := make(map[string]string)
+	copyStringMap(tempStructWatch, tempStruct)
+
+	tempStruct["resource_name"] = resourceName
+	tempStruct["watch_name"] = fmt.Sprintf("xray-watch-%d", randomInt())
+	tempStruct["policy_name_0"] = fmt.Sprintf("xray-policy-%d", randomInt())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() { testAccPreCheck(t) },
+		CheckDestroy: verifyDeleted(fqrn, func(id string, request *resty.Request) (*resty.Response, error) {
+			testCheckPolicyDeleted("xray_security_policy.security", t, request)
+			resp, err := testCheckWatch(id, request)
+			return resp, err
+		}),
+		ProviderFactories: testAccProviders(),
+		Steps: []resource.TestStep{
+			{
+				Config: executeTemplate(fqrn, allReposSinglePolicyWatchTemplate, tempStruct),
+				Check:  verifyXrayWatch(fqrn, tempStruct),
+			},
+		},
+	})
+}
+
+func TestAccWatch_allReposMultiplePolicies(t *testing.T) {
+	_, fqrn, resourceName := mkNames("watch-", "xray_watch")
+	tempStruct := make(map[string]string)
+	copyStringMap(tempStructWatch, tempStruct)
+
+	tempStruct["resource_name"] = resourceName
+	tempStruct["watch_name"] = fmt.Sprintf("xray-watch-%d", randomInt())
+	tempStruct["policy_name_0"] = fmt.Sprintf("xray-policy-%d", randomInt())
+	tempStruct["policy_name_1"] = fmt.Sprintf("xray-policy-%d", randomInt())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() { testAccPreCheck(t) },
+		CheckDestroy: verifyDeleted(fqrn, func(id string, request *resty.Request) (*resty.Response, error) {
+			testCheckPolicyDeleted("xray_security_policy.security", t, request)
+			testCheckPolicyDeleted("xray_license_policy.license", t, request)
+			resp, err := testCheckWatch(id, request)
+			return resp, err
+		}),
+
+		ProviderFactories: testAccProviders(),
+		Steps: []resource.TestStep{
+			{
+				Config: executeTemplate(fqrn, allReposMultiplePoliciesWatchTemplate, tempStruct),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(fqrn, "name", tempStruct["watch_name"]),
+					resource.TestCheckResourceAttr(fqrn, "description", tempStruct["description"]),
+					resource.TestCheckResourceAttr(fqrn, "watch_resource.0.type", tempStruct["watch_type"]),
```

Comment:
We should check for values for the policies here, right?

Reply:
Yeah, I've removed it on purpose when I had a sorting issue, forgot to add it back. Returning the verification back.